controller/lib/pid/pid.h

```diff
@@ -80,34 +85,31 @@ class PID {
   double GetKp();     // These functions query the pid for interal values.
   double GetKi();     //  they were created mainly for the pid front-end,
   double GetKd();     // where it's important to know what is actually
+  int GetSampleTime();
   int GetMode();      //  inside the PID.
   int GetDirection(); //
 
 private:
   void Initialize();
 
-  double dispKp; // * we'll hold on to the tuning parameters in user-entered
-  double dispKi; //   format for display purposes
-  double dispKd; //
-
   double kp; // * (P)roportional Tuning Parameter
   double ki; // * (I)ntegral Tuning Parameter
   double kd; // * (D)erivative Tuning Parameter
 
   int controllerDirection;
-  int pOn;
+  int pOn, dOn;
 
   double *myInput;  // * Pointers to the Input, Output, and Setpoint variables
   double *myOutput; //   This creates a hard link between the variables and the
   double
       *mySetpoint; //   PID, freeing the user from having to constantly tell us
                    //   what these values are.  with pointers we'll just know.
 
-  unsigned long lastTime;
-  double outputSum, lastInput;
+  unsigned long lastTime, lastSampleTime;
```

Comment:
Could you add a comment explaining the difference between these? Is lastTime the last time when Compute() was  actually called whereas lastSampleTime is when it _should have been_ called?

Reply:
You actually got them reverses, but it means this needs clarification. I will try a new approach to the time management, by renaming them nextSampleTime and lastUpdateTime, and having nextSampleTime be in the future instead of the past, which will probably make it clearer.